woocommerce/woocommerce-ios
# Product SKU input scanner release preparation

Completes the first feature of #2407 


### Description


We are ready to release the product SKU input scanner as the first feature of barcode scanner, and this PR contains a few updates before the release:

- Created a new feature flag `productSKUInputScanner` that always returns `true`, which is used for product SKU input scanner CTA. Since the more complicated inventory scanner (CTA on the Products tab) is still WIP, we want to release two features separately with two feature flags
- Added a feature switch for the feature (ref p91TBi-6Sv-p2#comment-7889)
- For both barcode scanner CTAs, added a check to only show the CTAs when camera is available on the device `UIImagePickerController.isSourceTypeAvailable(.camera)`
- Updated camera permission purpose string in `Info.plist` to include product SKU scanner

### Testing instructions


Let's simulate release configuration by testing with a device and manually disabling the `barcodeScanner` feature flag:

- In `DefaultFeatureFlagService`, return `false` for `barcodeScanner` to simulate release build (while `productSKUInputScanner` feature is always enabled)
- Launch the app on a device
- Go to the products tab --> the scan CTA should not be visible (behind `barcodeScanner` feature flag)
- Tap on a product whose SKU is editable
- Tap on the inventory/SKU row --> the scan CTA should not be visible because the feature switch is disabled by default
- Leave the product form and go to the My Store tab
- Go to Settings > Experimental Features --> there should be a section for Product SKU Scanner with a switch that is disabled by default
- Turn on the feature switch in the Product SKU Scanner section
- Leave the settings and go to the products tab
- Tap on a product whose SKU is editable
- Tap on the inventory/SKU row --> the scan CTA should be visible now
- Tap on the scan CTA --> it should navigate to the SKU input scanner

#### Other scenarios

@jaclync tests the following:

- [x] The feature switch remains the same state after switching stores
- [x] The feature switch remains the same state after logging out and in

### Screenshots


iOS camera permission alert | experimental features | scan CTA when feature switch is enabled
-- | -- | --
<img src="https://user-images.githubusercontent.com/1945542/146498944-7a0b9536-338c-4855-b8f7-68779bc44fba.png" width="300" /> | <img src="https://user-images.githubusercontent.com/1945542/147523983-ac9fe77d-e8fa-4c0a-9c6a-06de9ca42f70.png" width="300" /> | <img src="https://user-images.githubusercontent.com/1945542/147523985-c130b1dc-442f-41b7-9dea-5aa2aff34278.png" width="300" />



---
